kube-client/src/api/util.rs

```diff
@@ -34,6 +42,49 @@ impl Api<Node> {
     }
 }
 
+type CSRStatusType = &'static str;
+static APPROVED: CSRStatusType = "Approved";
+static DENIED: CSRStatusType = "Denied";
+
+impl Api<CertificateSigningRequest> {
+    /// Approve the specified CertificateSigningRequest.
+    pub async fn approve(&self, name: &str, pp: &PatchParams) -> Result<CertificateSigningRequest> {
+        self.patch_approval(name, pp, &APPROVED).await
+    }
+
+    /// Deny the specified CertificateSigningRequest.
+    pub async fn deny(&self, name: &str, pp: &PatchParams) -> Result<CertificateSigningRequest> {
+        self.patch_approval(name, pp, &DENIED).await
+    }
+
+    /// Partially update approval of the specified CertificateSigningRequest.
+    async fn patch_approval(
+        &self,
+        name: &str,
+        pp: &PatchParams,
+        approval_type: &CSRStatusType,
+    ) -> Result<CertificateSigningRequest> {
+        let patch = serde_json::json!({"status": CertificateSigningRequestStatus {
+            certificate:None,
+            conditions: Some(vec![CertificateSigningRequestCondition {
+                type_: approval_type.to_string(),
+                last_update_time: None,
+                last_transition_time: None,
+                message: Some(format!("{} {}", approval_type, "by kube-rs client")),
+                reason: Some("kube-rsClient".to_string()),
+                status: "True".to_string()
+            }])
```

Comment:
hm, while i do think these look like mostly sensible defaults. it's probably up to the controller doing this to attribute meaning to these fields. it's also slightly outside the purview of the api to be setting defaults here, it's something more normally done in the runtime crate.

personally, i think we should probably only have one method exposed from `Api<CertificateSigningRequest>`, and we should probably take a `CertificateSigningRequestStatus` directly. Maybe `::sign_respond`.

Reply:
I think it would be more straight-forward to just  call the fn `patch_approval`, WDYT?

Reply:
Ah, yeah, that sounds good :+1: